presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,228 @@
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.index.query.TermQueryBuilder;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.facebook.presto.elasticsearch.ElasticsearchErrorCode.ELASTICSEARCH_CONNECTION_ERROR;
+import static com.facebook.presto.spi.predicate.Marker.Bound.ABOVE;
+import static com.facebook.presto.spi.predicate.Marker.Bound.BELOW;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.elasticsearch.action.search.SearchType.QUERY_THEN_FETCH;
+
+public class ElasticsearchQueryBuilder
+{
+    private static final Logger LOG = Logger.get(ElasticsearchQueryBuilder.class);
+
+    private final Duration scrollTimeout;
+    private final int scrollSize;
+    private final TransportClient client;
+    private final int shard;
+    private final TupleDomain<ColumnHandle> tupleDomain;
+    private final List<ElasticsearchColumnHandle> columns;
+    private final String index;
+    private final String type;
+
+    public ElasticsearchQueryBuilder(List<ElasticsearchColumnHandle> columnHandles, ElasticsearchConnectorConfig config, ElasticsearchSplit split)
+    {
+        requireNonNull(columnHandles, "columnHandles is null");
+        requireNonNull(config, "config is null");
+        requireNonNull(split, "split is null");
+
+        ElasticsearchTableDescription table = split.getTable();
+        columns = columnHandles;
+        tupleDomain = split.getTupleDomain();
+        index = split.getIndex();
+        type = table.getType();
+        shard = split.getShard();
+        try {
+            Settings settings = Settings.builder()
+                    .put("client.transport.ignore_cluster_name", true)
+                    .build();
+            client = new PreBuiltTransportClient(settings)
+                    .addTransportAddress(new TransportAddress(InetAddress.getByName(split.getSearchNode()), split.getPort()));
+        }
+        catch (UnknownHostException e) {
+            throw new PrestoException(ELASTICSEARCH_CONNECTION_ERROR, format("Error connecting to Elasticsearch SearchNode: %s with port: %d", split.getSearchNode(), split.getPort()), e);
```

Comment:
`Error connecting to search node (%s:%d)`

Reply:
will fix